src/components/DpInput/DpInput.vue

```diff
@@ -30,124 +29,171 @@
       :required="required"
       :autocomplete="autocomplete !== '' ? autocomplete : null"
       :size="(size && size > 0) ? size : null"
-      v-model="currentValue"
-      @blur="$emit('blur', currentValue)"
-      @focus="$emit('focus')"
-      @input="$emit('input', currentValue)"
+      v-model="props.value"
+      @blur="emit('blur', $event.target.value)"
+      @focus="emit('focus')"
+      @input="(event) => { emit('update:modelValue', event.target.value); emit('input', event.target.value) }"
       @keydown.enter="handleEnter">
   </div>
 </template>
 
 <script setup lang="ts">
-import { computed, ref, toRefs, watch } from 'vue'
+import { computed } from 'vue'
 import { exactlengthHint, maxlengthHint, minlengthHint, prefixClass } from '~/utils'
 import DpLabel from '~/components/DpLabel'
 
-// Define an interface for the props
-interface Props {
-  ariaLabelledby?: string | null
-  autocomplete?: string
-  dataCounter?: string
-  dataCy?: string
-  dataDpValidateError?: string
-  dataDpValidateErrorFieldname?: string
-  dataDpValidateIf?: string
-  dataDpValidateShouldEqual?: string
-  disabled?: boolean | null
-  hasIcon?: boolean
-  id: string
-  label?: {
-    bold: boolean
-    hide: boolean
-    hint: string
-    text: string
-    tooltip: string
-  };
-  maxlength?: boolean | string | null
-  minlength?: boolean | string | null
-  name?: string
-  pattern?: string
-  placeholder?: string
-  preventDefaultOnEnter?: boolean
-  readonly?: boolean | null
-  required?: boolean | null
-  size?: number | null
-  type?: string
-  value?: string
-  width?: string
-}
+const props = defineProps({
+  ariaLabelledby: {
+    type: String,
+    required: false,
+    default: null
+  },
+
+  autocomplete: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataCounter: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataCy: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataDpValidateError: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataDpValidateErrorFieldname: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  dataDpValidateIf: {
+    type: String,
+    required: false,
+    default: ''
+  },
 
-// Use the interface with defineProps
-const props = defineProps<Props>();
-
-const emit = defineEmits(['blur', 'enter', 'focus', 'input'])
-
-const {
-  /**
```

Comment:
I think it is a good idea to keep the comments above each prop - while not every prop is documented, it is still a start.

Reply:
:innocent:

Reply:
got it. I restored the comments
746d40cce20c793f7daa4bd9ae83b9df3cfca6ec